cartographer/sensor/data.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2016 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_MAPPING_DATA_H_
+#define CARTOGRAPHER_MAPPING_DATA_H_
+
+#include <string>
+
+#include "cartographer/kalman_filter/pose_tracker.h"
+#include "cartographer/sensor/laser.h"
+#include "cartographer/transform/rigid_transform.h"
+
+namespace cartographer {
+namespace sensor {
+
+// This type is a logical union, i.e. only one type of sensor data is actually
+// filled in. It is only used for time ordering sensor data before passing it
+// on.
+struct Data {
+  enum class Type { kImu, kLaserFan3D, kOdometry };
+
+  struct Odometry {
+    transform::Rigid3d pose;
+    kalman_filter::PoseCovariance covariance;
```

Comment:
This is currently always set to a configured matrix. Remove the struct and just use a transform::Rigid3d?